torch/csrc/api/src/nn/modules/embedding.cpp

```diff
@@ -11,27 +11,72 @@
 namespace torch {
 namespace nn {
 
-EmbeddingOptions::EmbeddingOptions(int64_t count, int64_t dimension)
-    : count_(count), dimension_(dimension) {}
+EmbeddingOptions::EmbeddingOptions(int64_t num_embeddings, int64_t embedding_dim) : num_embeddings(num_embeddings), embedding_dim(embedding_dim) {}
 
-EmbeddingImpl::EmbeddingImpl(EmbeddingOptions options) : options(options) {
-  reset();
+EmbeddingImpl::EmbeddingImpl(EmbeddingOptions options): options(options) {
+  EmbeddingImpl::reset();
 }
 
 void EmbeddingImpl::reset() {
-  weight = register_parameter(
-      "weight", torch::empty({options.count_, options.dimension_}));
-  NoGradGuard guard;
-  weight.normal_(0, 1);
+  if (options.padding_idx() != c10::nullopt){
+    if(*options.padding_idx() > 0){
+      assert((*options.padding_idx() < *options.num_embeddings()) && "Padding_idx must be within num_embeddings");
+    }
+    else{
+      assert((*options.padding_idx() >= -(*options.num_embeddings())) && "Padding_idx must be within num_embedding");
+      options.padding_idx(*options.num_embeddings() + *options.padding_idx());
+    }
+  }
+  if (!options._weight().has_value()){
+    torch::nn::init.normal_(weight);
+  }
+  else{
+    weight = options._weight();
+    assert((weight.size(0) == options.num_embeddings()) && (weight.size(1) == embedding_dim) && "Shape of _weight does not match num_embeddings and embedding_dim");
+  }
 }
 
 void EmbeddingImpl::pretty_print(std::ostream& stream) const {
-  stream << "torch::nn::Embedding(count=" << options.count_
-         << ", dimension=" << options.dimension_ << ")";
+  stream << "torch::nn::Embedding(num_embeddings=" << options.num_embeddings()
+         << ", embedding_dim=" << options.embedding_dim();
+  if(options.padding_idx() != c10::nullopt){
+    stream << ",padding_idx=" << *options.padding_idx();
+  }
+  if(options.max_norm() != c10::nullopt){
+    stream << ",max_norm=" << *options.max_norm();
+  }
+  if(options.norm_type() != 2){
+    stream << ",norm_type=" << options.norm_type();
+  }
+  if(options.scale_grad_by_freq()){
+    stream << ",scale_grad_by_freq=" << options.scale_grad_by_freq();
+  }
+  if(options.sparse()){
```

Comment:
A note on formatting: in general we would want to write
```cpp
if (options.sparse()) {
```
instead of
```cpp
if(options.sparse()){
```
, and
```cpp
} else {
```
instead of
```cpp
}
else {
```
to be consistent with the formatting of the other parts of C++ API.